src/devenv-devShell.nix

```diff
@@ -18,7 +18,7 @@ pkgs.writeScriptBin "devenv" ''
 
   case $command in
     up)
-      procfilescript=${config.procfileScript}
+      procfilescript=$(nix build '.#devShells.${pkgs.stdenv.system}.default.config.procfileScript' --impure)
```

Comment:
Seems like a sledgehammer for fixing recursion errors. Here are some immediate issues:
* Delays the `nix build` call to the `devenv up` execution time and assumes that one does not need any additional nix flags. The `config.procfileScript` build might trigger new builds and those might require, say, a binary cache configuration that the user already passed in while running the initial `nix develop`.
* Won't work with [the Multiple shells setup](https://devenv.sh/guides/using-with-flakes/#modifying-your-flakenix-file:~:text=use%20flake%20.%20%2D%2Dimpure-,Multiple%20shells,-Defining%20multiple%20development).
* Assumes that our working directory is still in the same flake, so makes it impossible to run a devenv from a remote flake, like `nix develop github:me/myshell --impure`.
* Degrades the user experience by kicking off another round of potentially-lengthy `nix build` execution. Can be particularly bad for a workflow that relies on `nix build`ing the `devShell` somewhere else and embedding the resulting nix store hash into a script (along with a binary cache, for example).

I strongly recommend avoiding a sledgehammer like this and instead finding a way to resolve the root cause of the recursion.

Reply:
Those are fair points, could you open an issue? To fix this properly we need to decide on the trade offs.

The benefit of this approach is that we don't get into inf. recursive errors and that `devenv up` matches closer to what it does in devenv: evaluates and build on every CLI invocation - this is important as it's an expectation going between plain devenv and flakes integration.

Reply:
Sure, I'll gather my thoughts and open an issue as soon as I get the chance.

> `devenv up` matches closer to what it does in `devenv`

As a quick note; The more I think about this issue, the more it feels to me like this mini-`devenv` script doesn't belong in the nix-flake-based use case. When one uses nix flakes with the devenv project, one essentially replaces the `devenv` utility with the `nix` family of commands. So `devenv shell` becomes `nix develop`. Then it sounds to me like `devenv up` should become `nix run .#processes`. Extrapolating from this train of thought, maybe the right interface for nix flakes isn't `devenv.lib.mkShell`, but it's `devenv.lib.mkFlake` so that the mapping of the subcommands from `devenv` to nix flakes can be expressed all at once?